src/lightning_app/runners/cloud.py

```diff
@@ -326,20 +301,59 @@ def dispatch(
                     if existing_instance.status.phase != V1LightningappInstanceState.STOPPED:
                         raise RuntimeError("Failed to stop the existing instance.")
 
+            if app_config.cluster_id is not None:
+                # Verify that the cluster exists
+                list_clusters_resp = self.backend.client.cluster_service_list_clusters()
+                cluster_ids = [cluster.id for cluster in list_clusters_resp.clusters]
+                if app_config.cluster_id not in cluster_ids:
+                    if cluster_id:
+                        msg = f"You requested to run on cluster {cluster_id}, but that cluster doesn't exist."
+                    else:
+                        msg = (
+                            f"Your app last ran on cluster {app_config.cluster_id}, but that cluster "
+                            "doesn't exist anymore."
+                        )
+                    click.confirm(
+                        f"{msg} Do you want to run on Lightning Cloud instead?",
```

Comment:
Here the right ™️  thing to do would be to get the default cluster from the server (i.e. the cluster that corresponds to `None`) and then include that in the prompt. Not sure we have that notion though.

Reply:
Yeah, this should be updated when we have a way to figure out the default